### PR TITLE
Add fb cli front-end and grep utility

### PR DESCRIPTION
**Pull request by j3parker**

### CODE REVIEW

File: src/fb/Program.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		[Verb("grep", HelpText = "Search file buckets with regular expressions")]
+		private class GrepOptions : BaseOptions {
+			[Option('f', "file-pattern", HelpText = "Regular expression to filter searched file names", Default = "")]
+			public string FilenamePattern { get; set; }
+
+			[Option(Required = true, HelpText = "Regular expression to run on each line of every file" )]
+			public string Pattern { get; set; }
+		}
+
+		[Verb("gen-path-tree")]
```
Comment (j3parker) [inline src/fb/Program.cs]: This is just a dummy verb for now; CommandLineParser requires at least two verbs to function correctly. I'll wire up the other utility to this CLI front-end thing in a different PR.

File: src/fb/Program.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace fb {
+	class Program {
+		private class BaseOptions {
+			[Option(HelpText = "Folder containing filebucket git repos", Default = "//pdfs01/ea/devops/filebucket")]
+			public string BucketPath { get; set; }
+		}
+
+		[Verb("grep", HelpText = "Search file buckets with regular expressions")]
+		private class GrepOptions : BaseOptions {
+			[Option('f', "file-pattern", HelpText = "Regular expression to filter searched file names", Default = "")]
```
Comment (cesar-d2l) [inline src/fb/Program.cs]: `-f` is an option on `dotnet run` to use the target framework

Comment (j3parker) [inline src/fb/Program.cs]: Hm, the usual idiom for separating arguments is to put a `--` between one programs arguments and the next, e.g.

`dotnet run -c Release -- grep -f "foo" ...`

not sure if dotnet-cli works like that (you could also invoke the `fb` binary directly)
would it be helpful to rename this option regardless?

Comment (j3parker) [inline src/fb/Program.cs]: Yeah, looks like `--` should work https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-run?tabs=netcore2x (search "delimits" or "--")

**Pull request merged**